ui/src/message_popup/author/edit_question_page.jsx

```diff
@@ -14,12 +16,49 @@ export default React.createClass({
     questionId: React.PropTypes.string.isRequired
   },
 
+  getInitialState(){
+    return ({
+      loaded: false,
+      allDatabaseQuestions: {
+        currentQuestions: allQuestions,
+        archivedQuestions: allArchivedQuestions
+      }
+    });
+  },
+
+  componentDidMount(){
+    Api.questionsQuery().end(this.onQuestionsReceived);
+  },
+
+  onQuestionsReceived(err, response){
+    const questions = JSON.parse(response.text).questions;
+    if(questions !== undefined){
+      this.setState({loaded: true, allDatabaseQuestions: questions});
+      return;
+    }
+    this.setState({loaded: true});
+  },
+
   render(){
-    const question = withIndicator(withStudents([_.find(allQuestions, question => question.id.toString() === this.props.questionId)])[0]);
+    const {loaded, allDatabaseQuestions} = this.state;
+    const allQuestionsFlat =  this.state.allDatabaseQuestions.currentQuestions.concat(this.state.allDatabaseQuestions.archivedQuestions);
+    const originalQuestion = _.find(withStudents(allQuestionsFlat).map(question => withIndicator(question)), question => question.id.toString() === this.props.questionId);
     return (
-      <QuestionPage 
-        originalQuestion={question}
-        />
-      );
+      <div>
+        {!loaded &&
+          <QuestionPage
```

Comment:
Ah got it, thanks!  This makes sense then, or if it's just the `AppBar` that should render even during loading, you could pull that up to the top level component here.  The benefit being that loading states is removed from `QuestionPage` altogether, but the way you have it here looks good too.